tests/fv/fv_test.go

```diff
@@ -261,4 +260,169 @@ var _ = Describe("PolicyController", func() {
 			}, time.Second*10, 500*time.Millisecond).Should(Equal(0))
 		})
 	})
+
+	Describe("pod", func() {
+		It("labels are updated and active instance ids are respected", func() {
+			// Create a Pod
+			var wep api.WorkloadEndpoint
+			podName := "test-syncer-basic-pod"
+			pod := v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      podName,
+					Namespace: "default",
+					Labels: map[string]string{
+						"foo": "label1",
+					},
+				},
+				Spec: v1.PodSpec{
+					NodeName: "127.0.0.1",
+					Containers: []v1.Container{
+						v1.Container{
+							Name:    "container1",
+							Image:   "busybox",
+							Command: []string{"sleep", "3600"},
+						},
+					},
+				},
+			}
+			_, err := k8sClient.CoreV1().Pods("default").Create(&pod)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Update the Pod to 'running' and an IP.
+			pod.Status.PodIP = "192.168.1.1"
+			pod.Status.Phase = v1.PodRunning
+			_, err = k8sClient.Pods("default").UpdateStatus(&pod)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Mock the job of the CNI plugin by creating the wep in etcd, and assign the aii.
+			wep = api.WorkloadEndpoint{
+				Metadata: api.WorkloadEndpointMetadata{
+					Name:             "eth0",
+					Node:             pod.Spec.NodeName,
+					Orchestrator:     "k8s",
+					Workload:         "default." + podName,
+					ActiveInstanceID: "aii1",
+					Labels: map[string]string{
+						"foo": "label1",
+					},
+				},
+				Spec: api.WorkloadEndpointSpec{
+					InterfaceName: "eth0",
+				},
+			}
+			_, err = calicoClient.WorkloadEndpoints().Create(&wep)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Definitively trigger a policy controller cache update by updating a pod's labels
+			idMetadata := api.WorkloadEndpointMetadata{
+				Name:         "eth0",
+				Node:         pod.Spec.NodeName,
+				Orchestrator: "k8s",
+				Workload:     "default." + podName,
+			}
+			pod.ObjectMeta.Labels["foo"] = "label2"
+			_, err = k8sClient.CoreV1().Pods("default").Update(&pod)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Wait for the policy controller to update the web with the new labels
+			Eventually(func() Pair {
```

Comment:
Feels like this "Pair" concept is a bit roundabout.

Why not just have a function that returns an `error` and the logic inside the function returns a descriptive error when the condition isn't met?

If you're not happy with that approach and want to use the return struct approach we should rename the struct to be more descriptive - "Result" or something like like.